### PR TITLE
Calendar Improvements

### LINKED ISSUE #240: Implement Academic Calendar Business Objects Tests

### LINKED ISSUE #259: In the Academic Calendar format change from Details to Lectures
Details is very a very broad term. Lecture provides more clarity. 

```
{
   "creationDateTime": "20210421T204916Z",
   "retrievalDateTime": "20210421T204916Z",
   "school": {
      "name": "Instituto Superior Engenharia Lisboa",
      "acr": "ISEL"
   },
   "terms": [
      {
         "calendarTerm": "2019-2020-2",
         "interruptions": [],
         "evaluations": [],
         "lectures": [],
         "otherEvents": []
      }
   ]
}
```

### PR DESCRIPTION
**Pull request by rjcanto**
Changed BO and DTO tests. Closes #240 
Changed BOs and DTOs to reflect the name change from Details to Lectures. Closes #259

### CODE REVIEW

**Comment by rjcanto** [comment]
Also closes https://github.com/i-on-project/integration-data/issues/21

File: src/main/kotlin/org/ionproject/integration/domain/calendar/BusinessObjects.kt [hunk 1]
```diff
@@ -111,11 +106,58 @@ data class AcademicCalendar(
...
+                    descriptions.indexOf(
+                        descriptions.find {
+                            it.value.contains(
+                                regex.toRegex(
+                                    RegexOption.IGNORE_CASE
+                                )
+                            )
+                        }
+                    )
+                ].value
```
Comment (grimord) [inline src/main/kotlin/org/ionproject/integration/domain/calendar/BusinessObjects.kt]: maybe break this into a couple of statements? a bit hard to follow

Comment (rjcanto) [reply src/main/kotlin/org/ionproject/integration/domain/calendar/BusinessObjects.kt]: Improvement of `findFirstDateRangeFromEvent ` for clarity

**Review by grimord — APPROVED** [review]
